packages/block-editor/src/components/global-styles/typography-panel.js

```diff
@@ -257,33 +256,16 @@ export default function TypographyPanel( {
 
 	// Check if previous font style and weight values are available in the new font family
 	useEffect( () => {
-		const { fontStyles, fontWeights, isSystemFont } =
-			getFontStylesAndWeights( fontFamilyFaces );
-		const hasFontStyle = fontStyles?.some(
-			( { value: fs } ) => fs === fontStyle
-		);
-		const hasFontWeight = fontWeights?.some(
-			( { value: fw } ) => fw === fontWeight
-		);
-
-		// Try to set nearest available font weight
-		if ( ! hasFontWeight && fontWeight ) {
-			setFontAppearance( {
-				fontStyle,
-				fontWeight: findNearestFontWeight( fontWeights, fontWeight ),
-			} );
-		}
+		const { nearestFontStyle, nearestFontWeight } =
+			findNearestStyleAndWeight( fontFamilyFaces, fontStyle, fontWeight );
 
-		// Set the same weight and style values if the font family is a system font or if both are the same
-		if ( isSystemFont || ( hasFontStyle && hasFontWeight ) ) {
+		if ( nearestFontStyle && nearestFontWeight ) {
 			setFontAppearance( {
-				fontStyle,
-				fontWeight,
+				fontStyle: nearestFontStyle,
+				fontWeight: nearestFontWeight,
 			} );
-		}
-
-		// Reset font appearance if the font family does not have the selected font style
-		if ( ! hasFontStyle ) {
+		} else {
+			// Reset font appearance if there are no available styles or weights
 			resetFontAppearance();
 		}
 	}, [ fontFamily ] );
```

Comment:
There are some missing dependencies, which results in a linter warning.

I think it'd be good to make sure that's fixed or the rule is disabled for this line (with a comment mentioning why).

To fix it, it looks like `setFontAppearance` and `resetFontAppearance` would have to be wrapped in `useCallback`.

Reply:
I tried adding all of the dependencies but it looks like this causes an infinite loop, and I think it's OK to rely on only the `fontFamily` changing here as this is the main prop we want to watch in this case.

I've disabled the rule for this line with a comment explaining why: https://github.com/WordPress/gutenberg/pull/63215/commits/37963050aebcb6b0da3b42cf3139c5795421732d

I left the `useCallback` functions in for `setFontAppearance` and `resetFontAppearance` as they seem to work the same either way, and I'm guessing this might have some small performance benefits.